### PR TITLE
feat(api): submit survey endpoint

### PR DESCRIPTION
**Pull request by Nirajn2311**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52352

cc: @moT01

### CODE REVIEW

**Review by moT01 — APPROVED** [review]
This LGTM @Nirajn2311 🎉 (after conflicts fixed)

Only thing I noticed is that extra questions can be added in the request and they get saved to the database. Perhaps we need to validate that the questions and answers are only the specific values we expect. This exists on the current API as well, so I don't think we need to hold this PR up.

For future testers, here's a request body:

```
  {"surveyResults":{
    "title": "Foundational C# with Microsoft Survey",
    "responses": [
      {
        "question": "Please describe your role:",
        "response": "Experienced developer (more than 5 years experience)"
      },
      {
        "question": "Prior to this course, how experienced were you with .NET and C#?",
        "response": "Expert"
      }
    ]
  }}
  ```

**Comment by Nirajn2311** [comment]
TODO: Add ms-username property in response of `/get-session-user` in different PR

**Review by Sembauke — APPROVED** [review]

**Pull request merged**
